backend/controller/controller.go

```diff
@@ -815,6 +840,16 @@ func (s *Service) reconcileDeployments(ctx context.Context) (time.Duration, erro
 			})
 		}
 	}
+
+	// Clean up old failures, which can happen if a deployment was removed before successfully reconciling.
```

Comment:
ubernit: maybe move this comment up to the declaration of `oldFailures` so it's a bit more obvious up front